### ISSUE TITLE
[elastic2] No helpful output for debugging the problem: Error initializing message pipeline; plugin_name='java', location='#buffer:2:3'

### ISSUE THREAD

**Issue by grinapo** [syslog-ng/syslog-ng#2126]
## Version of syslog-ng
```
syslog-ng 3 (3.16.1)
Config version: 3.16
Installer-Version: 3.16.1
Revision: 3.16.1-1
Compile-Date: Jun 13 2018 21:42:55
Module-Directory: /usr/lib/syslog-ng/3.16
Module-Path: /usr/lib/syslog-ng/3.16
Available-Modules: afstomp,date,afprog,pseudofile,graphite,afsql,xml,cryptofuncs,appmodel,map-value-pairs,geoip2-plugin,geoip-plugin,dbparser,afamqp,linux-kmsg-format,afuser,sdjournal,mod-python,cef,afmongodb,disk-buffer,system-source,basicfuncs,confgen,afsmtp,kvformat,csvparser,json-plugin,syslogformat,afsocket,affile,add-contextual-data,hook-commands,tags-parser,redis,mod-java
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Systemd: on
```

## Platform
Debian stretch amd64

```
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-1~deb9u1-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

# Issue
**Error initializing message pipeline; plugin_name='java', location='#buffer:2:3'**

## Failure
I have tried to relate this to #1760 , #1140 and eventually #1188 , but since I do not have _any_ java error output I'm kind of stuck here. I see the same "syslog-ng-core.jar twice" (well, actually three times) but I cannot seem to be able to do anything about it, no matter how I fiddle with the related config files. 

This is the first time I have tried to use **elasticsearch2** (and have to say that the documentation needs some improvement which I will try to provide after I'm able to actually get it running), so it may be very well a configuration issue; however the problem is the absolute lack of information about what went wrong and where and how. I tried to raise the debug level to a point where I see any java output but no luck.  (As far as I see gdb chokes on java calls which is a refreshing effect for me but I have zero experience with that memory-eating beast called jvm, and I see no useful output apart from that it has crashed inside libjvm.so.)

```
[…]
[2018-06-20T12:39:34.439271] Log pattern database reloaded; file='/var/lib/syslog-ng/patterndb.xml', version='4', pub_date='2018-06-20'
[2018-06-20T12:39:34.439358] Processing the time zone file (32bit part); filename='/usr/share/zoneinfo/Europe/Budapest'
[2018-06-20T12:39:34.439384] Java machine new;
[2018-06-20T12:39:34.500028] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T12:39:34.500309] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T12:39:34.500402] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/log4j-1.2.16.jar;
[2018-06-20T12:39:34.500484] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-common.jar;
[2018-06-20T12:39:34.500596] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-v2.jar;
[2018-06-20T12:39:34.500706] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic.jar;
[2018-06-20T12:39:34.500809] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpclient-4.5.2.jar;
[2018-06-20T12:39:34.500937] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/gson-2.6.2.jar;
[2018-06-20T12:39:34.501071] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/guava-19.0.jar;
[2018-06-20T12:39:34.501233] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-codec-1.9.jar;
[2018-06-20T12:39:34.501335] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/jest-common-2.0.2.jar;
[2018-06-20T12:39:34.501448] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/jest-2.0.2.jar;
[2018-06-20T12:39:34.501541] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/slf4j-api-1.7.13.jar;
[2018-06-20T12:39:34.501637] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/slf4j-simple-1.7.13.jar;
[2018-06-20T12:39:34.501735] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpcore-4.4.4.jar;
[2018-06-20T12:39:34.501829] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-lang3-3.4.jar;
[2018-06-20T12:39:34.501927] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpcore-nio-4.4.4.jar;
[2018-06-20T12:39:34.502023] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpasyncclient-4.1.1.jar;
[2018-06-20T12:39:34.502140] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-logging-1.2.jar;
[2018-06-20T12:39:34.531440] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T12:39:34.534392] Error initializing message pipeline; plugin_name='java', location='#buffer:2:3'

```

## strace
It seems that strace thinks the same as gdb about SIGSEGV:
```
…
13:04:20.229919 mprotect(0x7f5376342000, 4096, PROT_READ) = 0
13:04:20.229943 mprotect(0x7f537654b000, 4096, PROT_READ) = 0
13:04:20.229964 munmap(0x7f537dda9000, 31048) = 0
13:04:20.229993 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/resources.jar", {st_mode=S_IFREG|0644, st_size=3509541, ...}) = 0
13:04:20.230020 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/rt.jar", {st_mode=S_IFREG|0644, st_size=64943117, ...}) = 0
13:04:20.230044 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/sunrsasign.jar", 0x7ffc59147520) = -1 ENOENT (No such file or directory)
13:04:20.230066 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jsse.jar", {st_mode=S_IFREG|0644, st_size=648802, ...}) = 0
13:04:20.230091 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jce.jar", {st_mode=S_IFREG|0644, st_size=96104, ...}) = 0
13:04:20.230115 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/charsets.jar", {st_mode=S_IFREG|0644, st_size=3135392, ...}) = 0
13:04:20.230139 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/jfr.jar", 0x7ffc59147520) = -1 ENOENT (No such file or directory)
13:04:20.230161 stat("/usr/lib/jvm/java-8-openjdk-amd64/jre/classes", 0x7ffc59147520) = -1 ENOENT (No such file or directory)
13:04:20.230183 open("/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/meta-index", O_RDONLY) = 14
13:04:20.230210 fstat(14, {st_mode=S_IFREG|0644, st_size=1922, ...}) = 0
13:04:20.230234 read(14, "% VERSION 2\n% WARNING: this file is auto-generated; do not edit\n...", 4096) = 1922
13:04:20.230274 read(14, "", 4096)      = 0
13:04:20.230312 close(14)               = 0
13:04:20.230332 mmap(NULL, 251658240, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7f536712a000
13:04:20.230354 mmap(0x7f536712a000, 2555904, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f536712a000
13:04:20.230377 mmap(NULL, 3932160, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7f5366d6a000
13:04:20.230398 mmap(0x7f5366d6a000, 40960, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f5366d6a000

13:04:20.230469 --- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=NULL} ---

13:04:20.230487 rt_sigreturn({mask=[]}) = 6
13:04:20.230604 mmap(0x80200000, 32210157568, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x80200000
13:04:20.230626 mmap(NULL, 62914560, PROT_NONE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_NORESERVE, -1, 0) = 0x7f536316a000
13:04:20.230646 mmap(0x7f5366d69000, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f5366d69000
13:04:20.230672 mmap(0x580100000, 703070208, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x580100000
13:04:20.230693 mmap(0x7f5365969000, 1376256, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f5365969000
…
```
which for me isn't really helpful, I was kind of aware that fishy things are going on around jvm.

## Configuration
The relevant config is:
```
@version: 3.16
@module mod-java
parser pattern_db {  db-parser(    file("/var/lib/syslog-ng/patterndb.xml")  );};
source s_remote2 { syslog( transport("udp") port(9401) ); };
destination d_elastic {
  elasticsearch2(
        index("syslog_ng-${YEAR}.${MONTH}.${DAY}")
        flush-limit(1)
        #client-lib-dir(/usr/share/elasticsearch/lib) 
        #client-lib-dir(.) # these commented out stuff were my investigation what options may be required to make it work.
        #server("elastic1")
        #port("9200")
        #resource("/etc/elasticsearch/elasticsearch.yml-crash-on-me-you-little-shit")
        #cluster_url("https://example.com:999")
        client-mode(node)
        cluster(elastic1)
        type("syslog")
        time-zone("Europe/Budapest")
        # template("$(format-json --scope rfc5424 --scope nv-pairs --exclude DATE --key ISODATE)")
        # custom_id(${UNIQID});
  );
};
log {        source(s_remote2); destination(d_elastic);};
```

**Comment by lbudai** [comment]
Hi,

when `client-mode` is set to `node` you have to specify the `client-lib-dir` option (which points to your elasticsearch libraries).
If it’s not require to use the `node` mode I’d suggest to use the `http` mode.

and I agree with you that there is room for documentation improvement.

(and in the future we want to support connecting to elastic via its REST API with a client implemented in the C prg. language)

**Comment by lbudai** [comment]
(and of course we will investigate the reported issue)

**Comment by grinapo** [comment]
@lbudai you may have noticed the commented out parts, which were kept epsecially to mark my path towards a UsefulBugreport™.

So here's with uncommented `client-lib-dir`:
```
[2018-06-20T19:46:49.419875] Java machine new;
[2018-06-20T19:46:49.488828] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T19:46:49.489201] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T19:46:49.489318] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/log4j-1.2.16.jar;
[2018-06-20T19:46:49.489449] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-common.jar;
[2018-06-20T19:46:49.489549] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-v2.jar;
[2018-06-20T19:46:49.489628] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic.jar;
[2018-06-20T19:46:49.489710] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpclient-4.5.2.jar;
[2018-06-20T19:46:49.489791] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/gson-2.6.2.jar;
[2018-06-20T19:46:49.489882] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/guava-19.0.jar;
[2018-06-20T19:46:49.489954] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-codec-1.9.jar;
[2018-06-20T19:46:49.490036] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/jest-common-2.0.2.jar;
[2018-06-20T19:46:49.490121] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/jest-2.0.2.jar;
[2018-06-20T19:46:49.490188] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/slf4j-api-1.7.13.jar;
[2018-06-20T19:46:49.490273] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/slf4j-simple-1.7.13.jar;
[2018-06-20T19:46:49.490339] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpcore-4.4.4.jar;
[2018-06-20T19:46:49.490405] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-lang3-3.4.jar;
[2018-06-20T19:46:49.490473] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpcore-nio-4.4.4.jar;
[2018-06-20T19:46:49.490571] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/httpasyncclient-4.1.1.jar;
[2018-06-20T19:46:49.490666] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/elastic-jest-client/commons-logging-1.2.jar;
[2018-06-20T19:46:49.490734] Add path to classpath: /usr/share/elasticsearch/lib/jackson-dataformat-yaml-2.8.10.jar;
[2018-06-20T19:46:49.490812] Add path to classpath: /usr/share/elasticsearch/lib/t-digest-3.2.jar;
[2018-06-20T19:46:49.490872] Add path to classpath: /usr/share/elasticsearch/lib/plugin-cli-6.3.0.jar;
[2018-06-20T19:46:49.490945] Add path to classpath: /usr/share/elasticsearch/lib/jna-4.5.1.jar;
[2018-06-20T19:46:49.490997] Add path to classpath: /usr/share/elasticsearch/lib/lucene-backward-codecs-7.3.1.jar;
[2018-06-20T19:46:49.491079] Add path to classpath: /usr/share/elasticsearch/lib/lucene-spatial-extras-7.3.1.jar;
[2018-06-20T19:46:49.491145] Add path to classpath: /usr/share/elasticsearch/lib/lucene-highlighter-7.3.1.jar;
[2018-06-20T19:46:49.491204] Add path to classpath: /usr/share/elasticsearch/lib/snakeyaml-1.17.jar;
[2018-06-20T19:46:49.491259] Add path to classpath: /usr/share/elasticsearch/lib/jackson-dataformat-cbor-2.8.10.jar;
[2018-06-20T19:46:49.491327] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-x-content-6.3.0.jar;
[2018-06-20T19:46:49.491387] Add path to classpath: /usr/share/elasticsearch/lib/log4j-1.2-api-2.9.1.jar;
[2018-06-20T19:46:49.491447] Add path to classpath: /usr/share/elasticsearch/lib/lucene-queries-7.3.1.jar;
[2018-06-20T19:46:49.491506] Add path to classpath: /usr/share/elasticsearch/lib/jackson-dataformat-smile-2.8.10.jar;
[2018-06-20T19:46:49.491569] Add path to classpath: /usr/share/elasticsearch/lib/HdrHistogram-2.1.9.jar;
[2018-06-20T19:46:49.491628] Add path to classpath: /usr/share/elasticsearch/lib/hppc-0.7.1.jar;
[2018-06-20T19:46:49.491697] Add path to classpath: /usr/share/elasticsearch/lib/log4j-api-2.9.1.jar;
[2018-06-20T19:46:49.491773] Add path to classpath: /usr/share/elasticsearch/lib/joda-time-2.9.9.jar;
[2018-06-20T19:46:49.491838] Add path to classpath: /usr/share/elasticsearch/lib/jopt-simple-5.0.2.jar;
[2018-06-20T19:46:49.491892] Add path to classpath: /usr/share/elasticsearch/lib/jackson-core-2.8.10.jar;
[2018-06-20T19:46:49.491956] Add path to classpath: /usr/share/elasticsearch/lib/spatial4j-0.7.jar;
[2018-06-20T19:46:49.492016] Add path to classpath: /usr/share/elasticsearch/lib/lucene-suggest-7.3.1.jar;
[2018-06-20T19:46:49.492101] Add path to classpath: /usr/share/elasticsearch/lib/lucene-queryparser-7.3.1.jar;
[2018-06-20T19:46:49.492174] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-6.3.0.jar;
[2018-06-20T19:46:49.492244] Add path to classpath: /usr/share/elasticsearch/lib/lucene-spatial3d-7.3.1.jar;
[2018-06-20T19:46:49.492319] Add path to classpath: /usr/share/elasticsearch/lib/lucene-spatial-7.3.1.jar;
[2018-06-20T19:46:49.492375] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-launchers-6.3.0.jar;
[2018-06-20T19:46:49.492431] Add path to classpath: /usr/share/elasticsearch/lib/lucene-sandbox-7.3.1.jar;
[2018-06-20T19:46:49.492557] Add path to classpath: /usr/share/elasticsearch/lib/lucene-join-7.3.1.jar;
[2018-06-20T19:46:49.492625] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-cli-6.3.0.jar;
[2018-06-20T19:46:49.492701] Add path to classpath: /usr/share/elasticsearch/lib/lucene-memory-7.3.1.jar;
[2018-06-20T19:46:49.492761] Add path to classpath: /usr/share/elasticsearch/lib/lucene-core-7.3.1.jar;
[2018-06-20T19:46:49.492814] Add path to classpath: /usr/share/elasticsearch/lib/lucene-misc-7.3.1.jar;
[2018-06-20T19:46:49.492866] Add path to classpath: /usr/share/elasticsearch/lib/lucene-grouping-7.3.1.jar;
[2018-06-20T19:46:49.492918] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-core-6.3.0.jar;
[2018-06-20T19:46:49.492987] Add path to classpath: /usr/share/elasticsearch/lib/plugin-classloader-6.3.0.jar;
[2018-06-20T19:46:49.493073] Add path to classpath: /usr/share/elasticsearch/lib/lucene-analyzers-common-7.3.1.jar;
[2018-06-20T19:46:49.493139] Add path to classpath: /usr/share/elasticsearch/lib/log4j-core-2.9.1.jar;
[2018-06-20T19:46:49.493201] Add path to classpath: /usr/share/elasticsearch/lib/jts-core-1.15.0.jar;
[2018-06-20T19:46:49.493256] Add path to classpath: /usr/share/elasticsearch/lib/elasticsearch-secure-sm-6.3.0.jar;
[2018-06-20T19:46:49.759039] Add path to classpath: /usr/lib/syslog-ng/3.16/java-modules/syslog-ng-core.jar;
[2018-06-20T19:46:49.785274] Error initializing message pipeline; plugin_name='java', location='#buffer:2:3'
```

It is extremely not clear which would be the preferred way to do it (elastic1? elastic2 using which method?), and I have guessed `node` would be the least wasteful. I'll check http mode but there I do not quite see why would I need a megaton of java instead of a slim http lib. :-)

**Comment by faxm0dem** [comment]
For http you only need the jest library

**Comment by lbudai** [comment]
Based on the attached client lib dir, your elastic version is 6.3.
Elasticseach and elasticsearch2 won't work with that version (it's because of incompatible internal api changes made by elasticsearch). So you should use the http mode (as faxm0dem mentioned, this is using the jest library, which is part of your installation).

**Comment by grinapo** [comment]
Okay. So let's sort out.

1. There is a documentation issue, namely which versions of Elastic are(n't) supported.
2. There is a documentation issue about the methods, where at least a short sentence would be nice about the comparative performance and the preference of the methods as well as their requirements (this is touched, but visibly not enough). [Or if you need enduser test results, mention it. :-)]
3. I'm not sure whether the final error could be more useful but it seems that there's been a SIGSEGV which got to the handler but it's not visible anywhere, including the debug log, which may not be intentional.

I'll check the alternative methods, obviously I don't expect them to show this error. Thank you!

**Comment by grinapo** [comment]
### Documentation ###
"HTTP mode. 
The syslog-ng OSE application sends messages over HTTP using the REST API of Elasticsearch, and uses the cluster-url() and **cluster()** options from the syslog-ng OSE configuration file. (…)

In version 3.10 and newer, you can list multiple servers in HTTP and HTTPS mode in the cluster-url() and **server()** options."

Is it intentional that the first is `cluster` and the second is `server`?

**Comment by MrAnno** [comment]
Hi,

> 3. I'm not sure whether the final error could be more useful but it seems that there's been a SIGSEGV which got to the handler but it's not visible anywhere, including the debug log, which may not be intentional.

It seems for me that the SIGSEGV comes from and handled by the Java VM, which is fine.

A quoute from [Stack Overflow](https://stackoverflow.com/questions/27241575/why-does-java-app-crash-in-gdb-but-runs-normally-in-real-life):

> Why does java app crash in gdb but runs normally in real life?

> Because it doesn't actually crash.
> Java uses speculative loads. If a pointer points to addressable memory, the load succeeds. Rarely the pointer does not point to addressable memory, and the attempted load generates SIGSEGV ... which java runtime intercepts, makes the memory addressable again, and restarts the load instruction.

The observed SIGSEGV signal occurs every time you start any of our Java destinations. Since it is not fatal, you can type `continue` in gdb to debug further.

**Comment by faxm0dem** [comment]
@grinapo `elasticsearch2(client-mode(http))` should work on ES 6.x

**Comment by lbudai** [comment]
@grinapo : as @faxm0dem  says, the http mode is working with ES 6.x

I'm trying to reproduce the issue, but before doing that I need some information:
how did you install syslog-ng ? (as I remember, I merged the version changes on 18th, june, so not sure how do you have syslog-ng 3.16 on 13th, june :) )

**Comment by grinapo** [comment]
I'll be off for a while so apologies if replies take longer.
@lbudai : It was installed from the wonderful repo of a _mysterious_ person: `deb http://download.opensuse.org/repositories/home:/laszlo_budai:/syslog-ng/Debian_9.0 ./`
I may have accidentally used a time machine, though.

@faxm0dem  as far as I remember it have erred just the same when using any java component, which seems just right, since it's the same jvm...

**Comment by lbudai** [comment]
Time machine is not the only option here.
It’s not related to your problem, it’s an issue for the so called mysterious person...

Ok, then I’ll check it on a freshly installed Debian 9.

**Comment by lbudai** [comment]
* ElasticSearch

I install ElasticSearch with docker (https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html)

```
 docker pull docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.0
```

run ElasticSearch:

```
docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.0
```


* syslog-ng (in a clean Debian 9.0 docker image)

I installed syslog-ng from my OBS repo:
```
 wget http://download.opensuse.org/repositories/home:laszlo_budai:/syslog-ng/Debian_9.0/Release.key
 apt-key add Release.key
  echo "deb http://download.opensuse.org/repositories/home:laszlo_budai:/syslog-ng/Debian_9.0 ./" > /etc/apt/sources.list.d/syslog-ng-obs.list
 apt update
 apt install -y syslog-ng-core syslog-ng-mod-elastic
```

  * configuration file
```
@version: 3.16
@include "scl.conf"

parser pattern_db {  db-parser(    file("/var/lib/syslog-ng/patterndb.xml")  );};

source s_remote2 { syslog( transport("tcp") port(9401) ); };

destination d_elastic {
  elasticsearch2(
        index("syslog_ng-${YEAR}.${MONTH}.${DAY}")
        flush-limit(1)
        cluster_url("http://172.17.0.2:9200")
        client-mode(http)
        type("syslog")
        time-zone("Europe/Budapest")
  );
};

log {
  source(s_remote2);
  destination(d_elastic);
  flags(flow-control);
};

```

  * run syslog-ng:

```
export LD_LIBRARY_PATH=/usr/lib/jvm/java-8-openjdk-amd64/jre/lib/amd64/server
syslog-ng -Fevdt
```

Then send logs with `loggen` (which is installed with syslog-ng-core as I remember):

```
bin/loggen --inet -P -n 10 172.17.0.1 9401

```

In the output of the `syslog-ng` I got this:
```
[2018-06-25T12:05:44.312122] http-outgoing-0 << HTTP/1.1 201 Created;
[2018-06-25T12:05:44.312142] http-outgoing-0 << Location: /syslog_ng-2018.06.25/syslog/Uc7VNmQBudIi_K0s4pPp;
[2018-06-25T12:05:44.312151] http-outgoing-0 << content-type: application/json\; charset=UTF-8;
[2018-06-25T12:05:44.312162] http-outgoing-0 << content-encoding: gzip;
[2018-06-25T12:05:44.312170] http-outgoing-0 << content-length: 175;

```

Based on the `Location`, it is to perform a request to the ElastiSearch,
(curl -X GET "${cluster-url}${Location}"):

```
curl -X GET "172.17.0.2:9200/syslog_ng-2018.06.25/syslog/Uc7VNmQBudIi_K0s4pPp"
```

The answer is:

```
{"_index":"syslog_ng-2018.06.25","_type":"syslog","_id":"Uc7VNmQBudIi_K0s4pPp","_version":1,"found":true,"_source":{"PROGRAM":"prg00000","PRIORITY":"info","PID":"1234","MESSAGE":"seq: 0000000009, thread: 0000, runid: 1529928344, stamp: 2018-06-25T14:05:44 PADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPADDPAD","ISODATE":"2018-06-25T14:05:44+02:00","HOST":"10.30.255.1","FACILITY":"auth","@timestamp":"2018-06-25T14:05:44+02:00"}}% 
```

**Comment by lbudai** [comment]
@grinapo : check my steps above and try to identify what could be the difference.

**Comment by lbudai** [comment]
@grinapo: any update on this?

**Comment by soubhik-c** [comment]
I am getting the same error with elasticsearch-6.4.1 and syslog-ng 3.17 in ubuntu 16.04. Any help is greatly appreciated.

Thanks

**Comment by gaborznagy** [comment]
Hi @grinapo,
Can you test your issue on the latest master, please? (it will be released in syslog-ng-3.20)
@furiel added improvements for debugging, see: https://github.com/balabit/syslog-ng/issues/1188#issuecomment-461737249
